@xen-orchestra/backups/disks/RemoteDisk.mjs

```diff
@@ -0,0 +1,81 @@
+import { RandomAccessDisk } from "@xen-orchestra/disk-transform"
+
+/**
+ * @typedef {import('@xen-orchestra/disk-transform').DiskBlock} DiskBlock
+ * @typedef {import('./RemoteDisk.mjs').DiskMetadata} DiskMetadata
+ * 
+ * @typedef {(Object)} DiskMetadata
+ * @property {(string)} uuid
+ * @property {(string)} parentId
+ * @property {(number)} virtualSize
+ * @property {(string)} parentLocation
+ */
+
+
+export class RemoteDisk extends RandomAccessDisk {
+    /**
+     * Abstract
+     * @returns {string}
+     */
+    getPath() { }
+
+    /**
+     * Abstract
+     * @returns {string}
+     */
+    getUuid() { }
+
+    /**
+     * Abstract
+     * @returns {number}
+     */
+    getMaxTableEntries() { }
+
+    /**
+     * Abstract
+     * Writes a full block.
+     * @param {number} index
+     * @param {Buffer} data
+     * @return {number
+     */
+    async writeBlock(index, data) { }
```

Comment:
there is an  type DiskBlock that is already widely used. DO you think iy's a good idea to use it here ?